### ISSUE TITLE
Add "Reduce LR On Plateau" scheduler  

### ISSUE THREAD

**Issue by haven-jeon** [dmlc/gluon-nlp#887]
## Description
The LR scheduler that reduce learning rate when a metric has stopped improving. 

## References
- https://arxiv.org/abs/1905.01969 uses this scheduler.
- https://keras.io/callbacks/#reducelronplateau
